claude,cursor: add cursor rules and gopls-mcp integration

benbardin opened the pull request:
Epic: none
Release note: None

benbardin committed:
claude,cursor: add cursor rules and gopls-mcp integration

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152676)
<!-- Reviewable:end -->

michae2 reviewed (commented):
Very cool. This is really interesting. I'm not quite sure how to review it, though. It seems like people might have different preferences about whether to have claude use gopls or not.

> @michae2 reviewed 1 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152676#-:-OZFZc38AqozwF3evEox:b-u4ys8g)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained

___
*[`CLAUDE.md` line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/152676#-OZFXRPh8u_xIfkcwlmP:-OZFXRPh8u_xIfkcwlmQ:b-abok33) ([raw file](https://github.com/cockroachdb/cockroach/blob/a111b35f93ce6e542a3d9e8eda20a43b75618e0f/CLAUDE.md#L80)):*
> ```Markdown
> ### Go Code Navigation and Analysis (gopls MCP)
> 
> When working with Go code in this repository, use the gopls MCP server tools for efficient code navigation and analysis. **Always prefer MCP tools over bash commands** (`grep`, `find`) for Go code analysis.
> ```

I think this might be controversial to put in everyone's CLAUDE.md. I'm not sure everyone is using gopls (especially people using GoLand). I wonder if this would be better as a section in https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/4668882962/Claude+Code?

What's the fallback like if gopls isn't running?

Will claude ever start gopls itself?


<!-- Sent from Reviewable.io -->

benbardin commented:
Sure! Reverted change to CLAUDE.md, added to https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/4668882962/Claude+Code instead.

As far as I know, goland/vscode/cursor/whatever will all do roughly the same thing, that is, start gopls in the background and use it for all of their nifty IDE code search/link features. So everyone would have an instance or two running now if you were to check :)

When cursor/claude start up with these files in place, it will ask you if you want to enable the MCP server - you say "yes" and you're good to go. It will call the startup command for you and interact with the server over stdio.

michae2 reviewed (commented):
I don't believe GoLand uses gopls.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152676#-:-OZGu26M0ZfST1gKT0_b:b-qssk3m)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @benbardin)



<!-- Sent from Reviewable.io -->

benbardin commented:
> I don't believe GoLand uses gopls.
Oh cool! Didn't realize that, learned something today :) Well, using this MCP is totally optional - it will pop up and ask when people start claude/cursor, and has to be enabled explicitly. Seem OK?

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @michae2 reviewed 3 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152676#-:-OZLmt4u1j_NvE2Au3cW:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @benbardin)



<!-- Sent from Reviewable.io -->

benbardin commented:
tftr!
